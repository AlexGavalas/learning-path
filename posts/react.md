---
title: 'React material'
date: '2021-11-11'
---

This is a list with material you can use to learn React. It will be updated frequently, so check it from time to time for new stuff.

You can start with the [tutorial from the React team](https://reactjs.org/tutorial/tutorial.html).
Some others tutorials are [this](https://www.react.express/) and [this](https://react-tutorial.app/).

[Here](https://learnreact.design/posts/what-is-react) is a more visual introduction to React.

Places to learn more about React:

-   [Best practices](https://alexkondov.com/tao-of-react/)
-   [Awesome React](https://github.com/enaqx/awesome-react)

### React relevant libraries

To learn about state management with Redux go [here](https://redux.js.org/tutorials/essentials/part-1-overview-concepts).

Check out the React Query library for data fetching [here](https://react-query.tanstack.com/).
