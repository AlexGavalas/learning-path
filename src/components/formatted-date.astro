---
import { format as formatDate, toDate, utcToZonedTime } from 'date-fns-tz';

export type Props = {
    dateString: string;
    timeZone?: string;
    format?: string;
};

const {
    dateString,
    format = 'd LLL yyyy',
    timeZone = 'Europe/Athens',
} = Astro.props;

const date = utcToZonedTime(toDate(dateString), timeZone);
---

<time datetime={date.toISOString()}>{formatDate(date, format)}</time>
