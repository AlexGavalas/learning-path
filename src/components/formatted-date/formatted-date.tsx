import { parseISO, format as formatDate } from 'date-fns';

interface DateProps {
    dateString: string;
    format?: string;
}

export const FormattedDate = ({
    dateString,
    format = 'd LLL yyyy',
}: DateProps) => {
    const date = parseISO(dateString);

    return <time dateTime={dateString}>{formatDate(date, format)}</time>;
};
