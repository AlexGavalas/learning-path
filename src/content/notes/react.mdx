---
title: React
created: '2021-11-11'
updated: '2023-06-16'
published: true
---

## Learn

-   [Official Docs](https://react.dev/learn)
-   [What is React](https://learnreact.design/posts/what-is-react)
-   [React Express](https://www.react.express/)
-   [React Tutorial](https://react-tutorial.app/)
-   [React Excercises](https://epic-react-exercises.vercel.app/)
-   [Best practices](https://alexkondov.com/tao-of-react/)
-   [Awesome React](https://github.com/enaqx/awesome-react)
-   [React 2025](https://docs.react2025.com/)
-   [React Philosophies](https://github.com/mithi/react-philosophies)
-   [React Resources](https://reactresources.com/)
-   [React Slides](https://marko-knoebl.github.io/slides/react-all-en.html)
-   [React Patterns](https://reactpatterns.com/)
-   [React Bits](https://github.com/vasanthk/react-bits)
-   [React Handbook](https://reacthandbook.dev)

---

## Libraries

Here are various maintained libraries divided per rough category.
They are either tried and found good or not tried but interesting.

### File uploading

-   [Filepond](https://github.com/pqina/react-filepond)
-   [React Uploady](https://github.com/rpldy/react-uploady)
-   [React Dropzone](https://github.com/react-dropzone/react-dropzone/)
-   [Uppy](https://github.com/transloadit/uppy)
-   [React Upload Box](https://github.com/cheersmas/react-upload-box)

===

### Stores

-   [Redux](https://redux.js.org/) ([Here](https://github.com/markerikson/react-redux-links) you can find more material
    regarding React/Redux and [here](https://github.com/redux-utilities/flux-standard-action) on the Redux action object).
    -   [Redux Toolkit](https://redux-toolkit.js.org/)
    -   [Redux Logic](https://github.com/jeffbski/redux-logic)
    -   [Redux Loop](https://github.com/redux-loop/redux-loop)
    -   [Redux Saga](https://github.com/redux-saga/redux-saga)
    -   [Redux Thunk](https://github.com/reduxjs/redux-thunk)
    -   [Reselect](https://github.com/reduxjs/reselect)
    -   [Rematch](https://github.com/rematch/rematch)
    -   [Hooked on Redux](https://github.com/harryhope/hooked-on-redux)
    -   [Rematch](https://rematchjs.org/)
-   [MobX](https://mobx.js.org/) (Find more stuff [here](https://github.com/mobxjs/awesome-mobx))
    -   [MobX State Tree](https://github.com/mobxjs/mobx-state-tree)
    -   [MobX Utils](https://github.com/mobxjs/mobx-utils)
-   [Recoil](https://recoiljs.org/)
-   [Zustand](https://github.com/pmndrs/zustand)
-   [Storeon](https://github.com/storeon/storeon)
-   [Effector](https://github.com/effector/effector)
-   [Little State Machine](https://github.com/bluebill1049/little-state-machine)
-   [Pullstate](https://github.com/lostpebble/pullstate)
-   [React Ridge State](https://github.com/web-ridge/react-ridge-state)
-   [Radioactive State](https://github.com/MananTank/radioactive-state)
-   [Jotai](https://github.com/pmndrs/jotai)
-   [Hookstate](https://github.com/avkonst/hookstate)
-   [Valtio](https://github.com/pmndrs/valtio)
-   [Juliette](https://github.com/markostanimirovic/juliette)
-   [SimpleR State](https://github.com/arnelenero/simpler-state)
-   [React Model](https://github.com/byte-fe/react-model)
-   [Elf](https://github.com/ngneat/elf)
-   [Teaful](https://github.com/teafuljs/teaful)
-   [Legend State](https://github.com/LegendApp/legend-state)
-   [Xoid](https://github.com/onurkerimov/xoid)
-   [nanostores](https://github.com/nanostores/nanostores)

#### Generic store utils

-   [React Tracked](https://github.com/dai-shi/react-tracked)
-   [Watchable](https://github.com/shortwave/watchable)

===

### Routing

-   [React Router](https://reactrouter.com/)
-   [@tanstack/router](https://github.com/TanStack/router)

===

### State Machines

-   [Robot](https://github.com/matthewp/robot)
-   [X-State](https://github.com/statelyai/xstate) (See [X-State Catalogue](https://xstate-catalogue.com/) for various implementations of state machines)

===

### React Hooks

Various hooks [here](https://usehooks.com/).

-   [React Use](https://github.com/streamich/react-use)
-   [Beautiful React Hooks](https://github.com/antonioru/beautiful-react-hooks)
-   [React Recipes](https://github.com/craig1123/react-recipes)
-   [Use Hooks TS](https://github.com/juliencrn/usehooks-ts)
-   [Use Sound](https://github.com/joshwcomeau/use-sound)
-   [Use Local Storage State](https://github.com/astoilkov/use-local-storage-state)
-   [Use Worker](https://github.com/alewin/useWorker)
-   [React Hotkeys Hook](https://github.com/JohannesKlauss/react-hotkeys-hook)
-   [Use Web Animations](https://github.com/wellyshen/use-web-animations)
-   [React Use Animations](https://github.com/useAnimations/react-useanimations)
-   [React Use Picture-in-Picture](https://github.com/DawChihLiou/react-use-pip)
-   [React Responsive](https://github.com/yocontra/react-responsive)
-   [React Use Auth](https://github.com/Swizec/useAuth)
-   [Use Eazy Auth](https://github.com/inmagik/use-eazy-auth)
-   [Rooks](https://github.com/imbhargav5/rooks)
-   [React Use Size](https://github.com/thomasthiebaud/react-use-size)

===

### Data Fetching

-   [React Query](https://github.com/tannerlinsley/react-query)
-   [SWR](https://swr.vercel.app/)
-   [Rest Hooks](https://resthooks.io/)
-   [React Fetch Hook](https://github.com/ilyalesik/react-fetch-hook)
-   [Axios Hooks](https://github.com/simoneb/axios-hooks)
-   [Use HTTP](https://github.com/ava/use-http)
-   [React Fetching Library](https://github.com/marcin-piela/react-fetching-library)
-   [React Async](https://docs.react-async.com/)
-   [GraphQL](https://graphql.org/) (A graphql client called [Relay](https://relay.dev/))
-   [Resourcerer](https://github.com/noahgrant/resourcerer)
-   [Hyper Fetch](https://github.com/BetterTyped/hyper-fetch)

===

### Real time communication

-   [React Use Websocket](https://github.com/robtaussig/react-use-websocket)

===

### Testing

-   [React Testing Library](https://testing-library.com/docs/react-testing-library/intro)
-   [React Perf Testing](https://github.com/keiya01/react-performance-testing)

See also <a href="/notes/testing">Testing</a>.

===

### Styling

#### CSS-in-JS

-   [Emotion](https://emotion.sh/docs/introduction)
-   [Styled Components](https://styled-components.com/)
-   [Styled System](https://styled-system.com/)
-   [Stiches](https://stitches.dev/)
-   [Griffel](https://github.com/microsoft/griffel)

#### Animation

-   [Framer Motion](https://www.framer.com/motion/)
-   [Animate.css](https://animate.style/)
-   [AnimXYZ](https://animxyz.com/)
-   [GSAP](https://greensock.com/docs/)
-   [React Spring](https://react-spring.io/)
-   [Motion](https://motion.dev/)
-   [Popmotion](https://popmotion.io/)
-   [Renature](https://formidable.com/open-source/renature/)
-   [Shifty](https://github.com/jeremyckahn/shifty/)
-   [Autoanimate](https://auto-animate.formkit.com/)

#### Utilities

-   [Classnames](https://github.com/JedWatson/classnames)
-   [clsx](https://github.com/lukeed/clsx)
-   [Polished](https://github.com/styled-components/polished)

===

### Components

#### Collections of components

A [list](https://github.com/brillout/awesome-react-components) with various components.

-   [Tailwind](https://tailwindui.com/)
-   [Preline](https://preline.co/)
-   [Gestalt](https://gestalt.pinterest.systems/installation)
-   [Base](https://baseweb.design/)
-   [Evergreen](https://evergreen.segment.com/)
-   [Headless](https://headlessui.dev/)
-   [Chakra](https://chakra-ui.com/)
-   [Material](https://mui.com/)
-   [Fluent](https://developer.microsoft.com/en-us/fluentui#/get-started/web)
-   [Stacks](https://stackoverflow.design/)
-   [Theme](https://theme-ui.com/)
-   [Aragon](https://ui.aragon.org/)
-   [Ant](https://ant.design/)
-   [Bumbag](https://www.bumbag.style/)
-   [Rainbow](https://react-rainbow.io/)
-   [Geist](https://geist-ui.dev/en-us)
-   [Radix](https://www.radix-ui.com/)
-   [Atomize](https://atomizecode.com/)
-   [Mantine](https://mantine.dev/)
-   [Vecha](https://github.com/vechai/vechaiui)
-   [Orbit](https://orbit.kiwi/)
-   [Next](https://nextui.org/)
-   [Formation](https://formation.fyi/)
-   [Elastic](https://github.com/elastic/eui)
-   [Daisy](https://daisyui.com/)
-   [Tremor](https://www.tremor.so)
-   [AriaKit](https://github.com/ariakit/ariakit)
-   [react-spectrum](https://github.com/adobe/react-spectrum)
-   [Rewind](https://github.com/rewindui/rewindui)

#### Dropdowns

-   [React Select](https://react-select.com/home)
-   [React Dropdown Select](https://sanusart.github.io/react-dropdown-select/)
-   [Choices](https://choices-js.github.io/Choices/)
-   [React Select Search](https://github.com/tbleckert/react-select-search)
-   [React Functional Select](https://github.com/based-ghost/react-functional-select)
-   [React Menu](https://github.com/szhsin/react-menu)
-   [Multiselect React Dropdown](https://github.com/srigar/multiselect-react-dropdown)
-   [Reaselect](https://github.com/reaviz/reaselct/)

#### Calendars

-   [React Datepicker](https://reactdatepicker.com/)
-   [React Nice Dates](https://reactnicedates.hernansartorio.com/)
-   [React Date Range](https://github.com/hypeserver/react-date-range)
-   [React Google Flight Datepicker](https://jslancerteam.github.io/react-google-flight-datepicker/)
-   [React Day Picker](https://github.com/gpbl/react-day-picker)

#### Checkboxes

-   [React Checkbox Tree](https://jakezatecky.github.io/react-checkbox-tree/)

#### Charts

-   [Highcharts](https://github.com/highcharts/highcharts-react)
-   [AM Charts](https://amcharts.com/)
-   [VisX](https://airbnb.io/visx/)
-   [Chart.js](https://github.com/reactchartjs/react-chartjs-2)
-   [Recharts](https://recharts.org/)
-   [D3](https://d3js.org/) (And an extensive [article](https://wattenberger.com/blog/react-and-d3) on React + D3).
-   [Apache Echarts](https://echarts.apache.org/en/index.html)
-   [Plotly](https://plotly.com/javascript/)
-   [Nivo](https://nivo.rocks/components)
-   [Victory](https://formidable.com/open-source/victory/)
-   [Frappe](https://frappe.io/charts)
-   [Billboard](https://naver.github.io/billboard.js/)
-   [Charts.css](https://chartscss.org/)
-   [Apex](https://github.com/apexcharts/react-apexcharts)
-   [Reaflow](https://github.com/reaviz/reaflow)
-   [Mafs](https://github.com/stevenpetryk/mafs)
-   [Polaris viz](https://github.com/Shopify/polaris-viz)
-   [Unovis](https://github.com/f5/unovis)
-   [Observable HQ Plot](https://github.com/observablehq/plot)

#### 3D graphics

-   [React Three Fiber](https://github.com/pmndrs/react-three-fiber) (Here is a React renderer for [Three.js](https://threejs.org/) and [this](https://market.pmnd.rs/) is a place to find models).
-   [Threeasy](https://threeasy.org/)
-   [Babylon](https://www.babylonjs.com/)

#### Loaders

-   [React Top Loading Bar](https://klendi.github.io/react-top-loading-bar/)
-   [React Content Loader](https://github.com/danilowoz/react-content-loader)
-   [React Loader Spinner](https://github.com/mhnpd/react-loader-spinner)
-   [React Rounder](https://github.com/furkanmavili/react-rounder)

#### Icons

-   [React Icons](https://github.com/react-icons/react-icons)
-   [Styled](https://github.com/styled-icons/styled-icons)
-   [Tabler](https://github.com/tabler/tabler-icons)
-   [Hamburger](https://hamburger-react.netlify.app/)
-   [Phosphor](https://github.com/phosphor-icons/phosphor-react)
-   [Akar](https://akaricons.com/)
-   [Iconoir](https://iconoir.com/)
-   [Flagpack](https://flagpack.xyz/)
-   [Basicons](https://basicons.xyz/)
-   [Radix Icons](https://icons.radix-ui.com/)

#### Tables

-   [React Table](https://react-table.tanstack.com/)
-   [Glide Data Grid](https://github.com/glideapps/glide-data-grid)
-   [React Super Responsive Table](https://github.com/coston/react-super-responsive-table)
-   [React Data Table Component](https://github.com/jbetancur/react-data-table-component)
-   [Ka Table](https://github.com/komarovalexander/ka-table)
-   [Mantine Data Table](https://github.com/icflorescu/mantine-datatable)

#### Notifications

-   [Cogo Toast](https://cogoport.github.io/cogo-toast/)
-   [Notyf](https://carlosroso.com/notyf/)
-   [React Toast Notifications](https://jossmac.github.io/react-toast-notifications/)
-   [Sweet Alert](https://sweetalert.js.org/)
-   [React Toastify](https://fkhadra.github.io/react-toastify/introduction)
-   [Reapop](https://github.com/LouisBarranqueiro/reapop)
-   [React Hot Toast](https://react-hot-toast.com/)
-   [Notistack](https://github.com/iamhosseindhv/notistack)
-   [sonner](https://github.com/emilkowalski/sonner)

#### Modals

-   [React Modal](https://github.com/reactjs/react-modal)
-   [React Cool Portal](https://github.com/wellyshen/react-cool-portal)
-   [Rodal](https://chenjiahan.github.io/rodal/)
-   [React Responsive Modal](https://github.com/pradel/react-responsive-modal)
-   [Nice Modal React](https://github.com/eBay/nice-modal-react)
-   [React Modal Sheet](https://github.com/Temzasse/react-modal-sheet)

#### Tooltips

-   [Popper](https://popper.js.org/)
-   [Tippy](https://atomiks.github.io/tippyjs/)
-   [React Hover](https://github.com/cht8687/react-hover)
-   [Nanopop](https://simonwep.github.io/nanopop/)
-   [Floating UI](https://github.com/atomiks/floating-ui)

#### Accordions

-   [React Collapsed](https://github.com/roginfarrer/react-collapsed)

#### Data Sheets

-   [React Spreadsheet](https://github.com/iddan/react-spreadsheet)
-   [React Datasheet](https://nadbm.github.io/react-datasheet/)
-   [Fortune Sheet](https://github.com/ruilisi/fortune-sheet)

#### Editors

-   [Draft.js](https://draftjs.org/)
-   [Slate](https://github.com/ianstormtaylor/slate)
-   [Edtr](https://edtr.io/)
-   [React Quill](https://github.com/zenoamaro/react-quill)
-   [Milkdown](https://github.com/saul-mirone/milkdown)
-   [Trix](https://github.com/basecamp/trix)

#### Scrollbars

-   [Overlay Scrollbars](https://github.com/KingSora/OverlayScrollbars)
-   [React Scrollbars Custom](https://github.com/xobotyi/react-scrollbars-custom)
-   [Smooth Scrollbar](https://github.com/idiotWu/smooth-scrollbar)

===

### Markdown

-   [React Markdown](https://github.com/remarkjs/react-markdown)

===

### Site Tour

-   [Intro.js](https://github.com/usablica/intro.js)
-   [Reactour](https://github.com/elrumordelaluz/reactour)
-   [Shepherd](https://github.com/shipshapecode/shepherd)

===

### PDF

-   [@react-pdf/renderer](https://react-pdf.org/)
-   [react-pdf](https://github.com/wojtekmaj/react-pdf)

===

### Timeline

-   [React Chrono](https://github.com/prabhuignoto/react-chrono)

===

### Storybook

-   [Storybook intro](https://storybook.js.org/docs/react/get-started/introduction)
-   [Storybook mobile](https://github.com/aholachek/storybook-mobile)
-   [Storybook Testing React](https://github.com/storybookjs/testing-react)
-   [Chromatic](https://www.chromatic.com/)
-   [Lost Pixel](https://github.com/lost-pixel/lost-pixel)

Alternatives to storybook are

-   [Ladle](https://www.ladle.dev/)
-   [whyframe](https://whyframe.dev/)
-   [React Styleguidist](https://react-styleguidist.js.org/)

===

### RxJS

-   [React RxJS](https://react-rxjs.org/)

See also <a href="/notes/javascript#rxjs">RxJS in Javascript</a>

===

### Utilities

#### Date

-   [Date fns](https://date-fns.org/)
-   [Luxon](https://moment.github.io/luxon/)
-   [Day.js](https://day.js.org/)
-   [Spacetime](https://github.com/spencermountain/spacetime)

#### Forms

-   [React Hook Form](https://react-hook-form.com/)
-   [React Final Form](https://github.com/final-form/react-final-form)
-   [React Form](https://github.com/tannerlinsley/react-form)
-   [React Zorm](https://github.com/esamattis/react-zorm)
-   [MobX React Form](https://foxhound87.github.io/mobx-react-form/)
-   [Formik](https://formik.org/)
-   [Uniforms](https://uniforms.tools/)
-   [React JSON Schema Form](https://github.com/rjsf-team/react-jsonschema-form)
-   [Typed React Form](https://github.com/CodeStix/typed-react-form)
-   [Use Reactive Form](https://github.com/Michaeladze/useReactiveForm)
-   [Use Form](https://github.com/use-form/use-form)

#### Lazy Load

-   [Loadable Components](https://loadable-components.com/)

#### Data Sheets

-   [Papa Parse](https://www.papaparse.com/)

===

### i18n

-   [React i18next](https://react.i18next.com/), and a util [site](https://translate.i18next.com/) to translate JSON resources.
-   [Lingui](https://lingui.js.org/)
-   [FBT](https://facebook.github.io/fbt/)
-   [Rosetta](https://github.com/lukeed/rosetta)
-   [Talkr](https://github.com/DoneDeal0/Talkr)
-   [Typesafe i18n](https://github.com/ivanhofer/typesafe-i18n)
-   [Schummar Translate](https://github.com/schummar/schummar-translate)

===

### Router

-   [React Router](https://reactrouter.com/)
-   [React Chicane](https://github.com/zoontek/react-chicane)

===

### Events

-   [React Cool onClick Outside](https://github.com/wellyshen/react-cool-onclickoutside)
-   [Why Did You Render](https://github.com/welldone-software/why-did-you-render)
-   [Tiny Keys](https://jamiebuilds.github.io/tinykeys/)
-   [Kbar](https://kbar.vercel.app/)
-   [Ninja Keys](https://github.com/ssleptsov/ninja-keys)

===

### Layout

-   [React Mosaic](https://github.com/nomcopter/react-mosaic)
-   [React Grid Layout](https://github.com/react-grid-layout/react-grid-layout)
-   [React Snuggle](https://github.com/Significa/react-snuggle)

===

### Maps

-   [Kepler](https://kepler.gl/)
-   [React Map GL](https://visgl.github.io/react-map-gl/)

===

### Validation

-   [Vest](https://github.com/ealush/vest)

===

### Fonts

-   [Fontsource](https://fontsource.org/)

---

### Various

-   [React Particle Image](https://github.com/malerba118/react-particle-image)
-   [React Curved Arrow](https://github.com/nickjanssen/react-curved-arrow)
-   [React Autosuggest](https://github.com/moroshko/react-autosuggest)
-   [React Audio Player](https://github.com/lhz516/react-h5-audio-player)
-   [Spectacle](https://formidable.com/open-source/spectacle/)
-   [React Flow](https://github.com/wbkd/react-flow)
-   [React Digraph](https://github.com/uber/react-digraph)
-   [Beautiful React Diagrams](https://github.com/antonioru/beautiful-react-diagrams)
-   [React Easy Edit](https://github.com/giorgosart/react-easy-edit)
-   [React Page Visibility](https://github.com/pgilad/react-page-visibility)
-   [React Shimmer](https://github.com/gokcan/react-shimmer)
-   [Perfect Arrows](https://github.com/steveruizok/perfect-arrows)
-   [Chonky](https://github.com/TimboKZ/Chonky)
-   [PWA Install](https://github.com/pwa-builder/pwa-install)
-   [TS Particles](https://github.com/matteobruni/tsparticles)
-   [Comlink](https://github.com/GoogleChromeLabs/comlink)
-   [Automerge](https://github.com/automerge/automerge)
-   [Simple Peer](https://github.com/feross/simple-peer)
-   [React Colorful](https://github.com/omgovich/react-colorful)
-   [React Speech Recognition](https://github.com/JamesBrill/react-speech-recognition)
-   [DnD Kit](https://github.com/clauderic/dnd-kit)
-   [React DnD](https://github.com/react-dnd/react-dnd)
-   [React Email Editor](https://github.com/unlayer/react-email-editor)
-   [React Email](https://github.com/zenorocha/react-email)
-   [Mailing](https://github.com/successor-software/mailing)
-   [Remotion](https://www.remotion.dev/)
-   [Inspx](https://github.com/raunofreiberg/inspx)
-   [Svgr](https://github.com/gregberge/svgr)
-   [React Tags](https://github.com/react-tags/react-tags)
-   [Tagify](https://github.com/yairEO/tagify)
-   [React Stepzilla](https://github.com/newbreedofgeek/react-stepzilla)
-   [React Error Boundary](https://github.com/bvaughn/react-error-boundary)
-   [React Webcam](https://github.com/mozmorris/react-webcam)
-   [Color](https://github.com/Qix-/color)
-   [Colord](https://colord.omgovich.ru/)
-   [React Fire](https://github.com/FirebaseExtended/reactfire)
-   [React Cool Inview](https://github.com/wellyshen/react-cool-inview)
-   [React Focus Lock](https://github.com/theKashey/react-focus-lock)
-   [Fuite](https://github.com/nolanlawson/fuite)
-   [React Arborist](https://github.com/brimdata/react-arborist)
-   [React Flow](https://reactflow.dev/)
-   [Cmdk](https://cmdk.paco.me/)
-   [Kmenu](https://github.com/harshhhdev/kmenu)
-   [Electron React Boilerplate](https://electron-react-boilerplate.js.org/)
-   [Electron Forge](https://www.electronforge.io/)
-   [Neutralino](https://github.com/neutralinojs/neutralinojs)
-   [sequential-workflow-designer](https://github.com/nocode-js/sequential-workflow-designer)

---

### Cheatsheets

Quick info regarding React/Typescript can be found [here](https://react-typescript-cheatsheet.netlify.app/) and [here](https://github.com/typescript-cheatsheets/react).

---

### Interview Questions

You can find various React questions [here](https://medium.com/@stevenjinyi/study-sheet-for-react-interview-questions-2020-2fe25b8fa316), [here](https://dev.to/scrimba/react-interview-questions-to-expect-in-2021-with-answers-dfl), [here](https://github.com/sudheerj/reactjs-interview-questions), [here](https://dev.to/aviyel/300-react-interview-questions-4dag) and [here](https://code.coursesity.com/react-interview-questions).
