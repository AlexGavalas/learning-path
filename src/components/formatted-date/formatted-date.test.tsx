import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';

import { FormattedDate } from './formatted-date';

const mockDate = new Date('2022-01-01');
const mockDateISO = mockDate.toISOString();

describe('<FormattedDate />', () => {
    it('renders', () => {
        const { container } = render(
            <FormattedDate dateString={mockDateISO} />,
        );

        expect(container).toMatchSnapshot();
    });

    it('is accessible', async () => {
        const { container } = render(
            <FormattedDate dateString={mockDateISO} />,
        );

        const a11yResults = await axe(container);

        expect(a11yResults).toHaveNoViolations();
    });

    it('uses the passed format', () => {
        render(<FormattedDate dateString={mockDateISO} format="dd/MM/yyyy" />);

        expect(screen.getByText('01/01/2022')).toBeInTheDocument();
    });
});
