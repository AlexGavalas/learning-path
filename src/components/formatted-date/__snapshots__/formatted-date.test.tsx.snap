// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedDate /> renders 1`] = `
<div>
  <time
    datetime="2022-01-01T00:00:00.000Z"
  >
    1 Jan 2022
  </time>
</div>
`;
